### ISSUE TITLE
Executing a script from the startup_once hook does not work without adding a delay

### ISSUE THREAD

**Issue by TinyAtoms** [qtile/qtile#3768]
### The issue:

# problem 

I have the following in my `config.py` 
```
@hook.subscribe.startup_once
def autostart():
    home = os.path.expanduser('~/.config/qtile/autostart.sh')
    subprocess.Popen([home])
```
and the following in the `autostart.sh`

```
#!/bin/bash
waybar &
```
Waybar does not start.

The following has worked
```
@hook.subscribe.startup_once
async def autostart():
    await asyncio.sleep(3)
    # sleep(3)
    home = os.path.expanduser('~/.config/qtile/autostart.sh')
    subprocess.Popen([home])
```
(I can do it synchronously too, but that does give a blank screen while it's sleeping)
I have also tried it with a variety of different programs besides waybar, such as telegram-desktop, discord-canary and whatsap-for-linux. None of them start if I do not wait first. 
Qtile log does not show any errors. 
I have tried logging the output by doing stuff like `waybar > autostart.log` in the startup script, but that log also doesn't get anything.
I have tried logging the output of subprocess.Popen directly, but that makes qtile freeze.

## Environment

Distro: Arch linux
qtile 0.21.0 (The issue is not here in https://aur.archlinux.org/packages/qtile-wayland-git  0.21.0.r89.gc8d3aad1-1 for neither wayland or x11)
wlroots 0.15.1-6
wayland 1.21.0-1
wayland-protocols 1.26-1
xorg-xwayland 22.1.3-1
python-pywayland 0.4.14-1

 

### Required:

- [X] I have searched past issues to see if this bug has already been reported.

**Comment by elParaguayo** [comment]
That's odd.

Does this happen with the default config? What about on x11 (using something other than waybar!)?

**Comment by TinyAtoms** [comment]
I can't replicate the issue anymore.
It used to happen with qtile installed from the arch repos, but due to an issue, I switched over to  https://aur.archlinux.org/packages/qtile-wayland-git 
I just tried switching back to the the arch repo one by 
```
yay -R qtile-wayland-git
sudo pacman -Qtdq  sudo| pacman -Rns - 
sudo pacman -S qtile
``` 
I can't do extensive testing, as this is my daily driver now.
From what I do remember:
1. it used to happen with a pretty bog standard config. I only changed some keybindings before trying to add waybar with the hook. 
2. I did try it with other applications besides waybar (telegram-for-desktop, discord-canary, whatsapp-for-linux), and it worked for none of them
3. I do not remember testing it with x11
